api/models/shared_abstract/group_or_person.py

```diff
@@ -9,6 +9,7 @@
 class GroupOrPerson(models.Model):
     """Model holding attributes that are common for every person and group."""
 
+    legacy_id = models.IntegerField(null=True)
```

Comment:
Not sure what is more safe, to have unique constraint or not, for now I think it is okay

Reply:
Here's the thing: one person can have multiple IDs: e.g. one as teacher, one as coordinator. That's why we switched to `PersonalInfo` in the first place: to create a unique ID per person, not per role of person. So there cannot be just one `.legacy_id`.

That's why I initially suggested creating `.legacy_sid` in `Student`, `.legacy_tid` in `Teacher` etc. And then I forgot my own motivation when we talked about moving the legacy ID into the abstract model.

Reply:
So you want to create a single personal info for one "person", not sure it is possible, how we map between different roles?

Reply:
Any way we will have multiple personal infos for a single person

Reply:
> Any way we will have multiple personal infos for a single person

No, it's the other way round 😸 

Each human individual is described by only one `PersonalInfo` object. This object is then used as a primary key for `Person` abstract class, e.g. to create `Student`s, `Coordinator`s and teachers. The [way it's created](https://github.com/samanthasgroup/django-webapps/blob/master/api/models/shared_abstract/person.py#L13C21-L18) ensures that for each `PersonalInfo` there can be only one `Student` stored in `.as_student` attribute, only one `Coordinator` as `.as_coordinator` etc.

Reply:
Actually, this means that we should tweak the `BasePopulator` and use `.get_or_create()` instead of `.create` in [this method](https://github.com/samanthasgroup/django-webapps/blob/master/django_webapps/scripts/db_population/base_populator.py#L115-L130) (maybe it could be renamed too). I guess I should create a separate issue for this.

Reply:
I meant exactly that, how do you identify a duplicate?

Reply:
If there any rule to use exactly one email or name?

Reply:
It does make sense to not create a separate Personal info if email is duplicated

Reply:
And we can filter by unique emails from the input CSV, we just need clarification from the owners that emails are unique (usually at least)

Reply:
> is there any rule to use exactly one email or name?

The combination of `first_name`, `last_name` and `email` must be unique.

Since people can e.g. register their children, `email` alone is not good for uniqueness test.